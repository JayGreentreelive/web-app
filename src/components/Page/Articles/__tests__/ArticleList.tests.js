import React from 'react'
import { MockedProvider } from '@apollo/react-testing'
import { act, render } from '@testing-library/react'
import wait from 'waait'
import renderer from "react-test-renderer"
import { ARTICLE_LIST_MOCKS, ARTICLE_LIST_ERROR_MOCKS } from './mocks'

import { ArticleList } from '../'

let component = null

describe("ArticleList", () => {

    it("renders without crashing", () => {
        console.log({ ARTICLE_LIST_MOCKS })


        act(() => {
            render(
                <MockedProvider mocks={ARTICLE_LIST_MOCKS} addTypename={false}>
                    <ArticleList />
                </MockedProvider>
            )
        })
    })

    it("renders a list of articles", async () => {
        act(() => {
            component = render(
                <MockedProvider mocks={ARTICLE_LIST_MOCKS} addTypename={false}>
                    <ArticleList />
                </MockedProvider>
            )
        })

        await wait(0) // waits for response

        const { container } = component
        expect(container).toMatchSnapshot()
    })

    it("renders the loading state", () => {
        act(() => {
            component = render(
                <MockedProvider mocks={[]}>
                    <ArticleList />
                </MockedProvider>
            )
        })

        const { container } = component
        expect(container).toMatchSnapshot()
    })

    it("renders the error state", async () => {
        act(() => {
            component = render(
                <MockedProvider mocks={ARTICLE_LIST_ERROR_MOCKS}>
                    <ArticleList />
                </MockedProvider>
            )
        })

        await wait(0) // waits for response

        const { container } = component
        expect(container).toMatchSnapshot()
    })
})
