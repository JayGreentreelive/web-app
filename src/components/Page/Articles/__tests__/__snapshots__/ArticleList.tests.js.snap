// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders a list of articles 1`] = `
<div>
  <div
    class="container py-6"
  >
    <div
      class="row"
    >
      <div
        class="col text-center"
      >
        <h3>
          Click on an article below to read it
        </h3>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-6 col-md-3 p-3"
      >
        <div
          class="card h-100"
        >
          <div
            class="embed-responsive embed-responsive-16by9  card-img-top"
          >
            <div
              class="fill d-flex justify-content-center align-items-center"
            >
              <div
                class="spinner-border text-primary"
                role="status"
              >
                <span
                  class="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Article 1
            </h5>
            <p
              class="card-text"
            >
              READ THIS GUYS!...
            </p>
            <a
              href="articles/article-1/"
            >
              Read article
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-6 col-md-3 p-3"
      >
        <div
          class="card h-100"
        >
          <div
            class="embed-responsive embed-responsive-16by9  card-img-top"
          >
            <div
              class="fill d-flex justify-content-center align-items-center"
            >
              <div
                class="spinner-border text-primary"
                role="status"
              >
                <span
                  class="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Article 2
            </h5>
            <p
              class="card-text"
            >
              READ THIS GUYS!...
            </p>
            <a
              href="articles/article-2/"
            >
              Read article
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleList renders the error state 1`] = `
<div>
  <h3
    class="text-center text-danger"
  >
    There was an error loading this content. Try refreshing the page.
  </h3>
</div>
`;

exports[`ArticleList renders the loading state 1`] = `
<div>
  <div
    class="fill d-flex justify-content-center align-items-center"
  >
    <div
      class="spinner-border text-primary"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
